ci/Makefile

```diff
@@ -1,11 +1,36 @@
-.PHONY: run restart-proxy restart-batch-proxy restart-all-proxies
+.PHONY: check-vars run restart-proxy restart-batch-proxy restart-all-proxies
 .PHONY: setup-conda-env build-hail-ci push-hail-ci test-locally
 .PHONY: run-service deploy
 
-PROJECT = $(shell gcloud config get-value project)
+PROJECT := $(shell gcloud config get-value project)
+
+# Grabs details of first instance matching HAIL_CI_INSTANCE
+DETAILS := $(shell gcloud compute instances list --format='table(NAME,STATUS,tags.list(),EXTERNAL_IP,ZONE)' | tail -n +2 | grep "$(HAIL_CI_INSTANCE)" --color=never | grep RUNNING | head -n 1 | tr -s ' ' '\n' )
+INSTANCE := $(word 1, $(DETAILS))
+EXTERNAL_IP := $(word 4, $(DETAILS))
+ZONE := $(word 5, $(DETAILS))
 
 HAIL_CI_LOCAL_BATCH_PORT ?= 8888
 
+# Must not be indented for conditions to work
+check-vars:
```

Comment:
rather than all these checks and this make target, I'm inclined to define the variables above as follows:
```
ifeq ($(words $(EXTERNAL_IP_AND_ZONE)),2)
EXTERNAL_IP := $(word 1, $(EXTERNAL_IP_AND_ZONE))
ZONE := $(word 2, $(EXTERNAL_IP_AND_ZONE))
else
EXTERNAL_IP = $(error "instance look up failed, $(HAIL_CI_INSTANCE), $(EXTERNAL_IP_AND_ZONE)")
ZONE = $(EXTERNAL_IP)
endif
```
A single equals sign is a lazy variable (properly called a "recursively expanded variable"), so it's definition is evaluated at use-site, not definition-site. Moreover, zone will recursively try to load EXTERNAL_IP which will print the error. I think this is a safer solution than `check-vars` because anyone who changes this file needs to make sure check-vars protects any reference to ZONE and EXTERNAL_IP. With this definition, make will always error if they're not properly defined.

Reply:
It worked well, thanks!